src/rect.rs

```diff
@@ -420,6 +420,47 @@ impl Rect {
     pub fn to_rounded_rect(self, radius: f64) -> RoundedRect {
         RoundedRect::from_rect(self, radius)
     }
+
+    /// Returns the largest possible `Rect` that is fully contained in this rect with the fixed
+    /// aspect ratio given by `aspect_ratio`.
+    ///
+    /// The resulting rectangle will be centered if it is smaller than the input rectangle.
+    ///
+    /// For the special case where the aspect ratio is `1.0`, the resulting `Rect` will be square.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use kurbo::Rect;
+    /// let outer = Rect::new(0.0, 0.0, 10.0, 20.0);
+    /// let inner = outer.fix_aspect_ratio(1.0);
+    /// // The new `Rect` is a square centered at the center of `outer`.
+    /// assert_eq!(inner, Rect::new(0.0, 5.0, 10.0, 15.0));
+    /// ```
+    ///
+    pub fn fix_aspect_ratio(&self, aspect_ratio: f64) -> Rect {
+        let (width, height) = (self.width(), self.height());
+        let self_aspect = height / width;
+
+        if self_aspect == aspect_ratio {
```

Comment:
This seems like a place where floating point error could come up, although the downside is negligible.

Reply:
Is there a `fuzzy_eq` or similar?

Reply:
I've done `(x - y).abs() < 1e-6` for now - is there a standard negligible error you use?

Reply:
no, you'd have to just pick a threshold and check against that. We do really want to have some sort of comprehensive fuzzy_eq implementation but we haven't done it yet.

Reply:
I've found [float-cmp](https://crates.io/crates/float-cmp) to be pretty good, but admittedly my floating point understanding is superficial.